filters/auth/oauth.go

```diff
@@ -173,6 +235,37 @@ func jsonGet(url *url.URL, auth string, doc interface{}) error {
 	return d.Decode(doc)
 }
 
+// jsonPost requests url with access token in the body, if auth was given and writes into doc.
+func jsonPost(u *url.URL, auth string, doc *tokenIntrospectionInfo) error {
+	if auth == "" {
+		return fmt.Errorf("invalid without auth")
+	}
+
+	body := url.Values{}
+	body.Add("token", auth)
+
+	rsp, err := http.PostForm(u.String(), body)
+	if err != nil {
+		return err
+	}
+
+	defer rsp.Body.Close()
+	if rsp.StatusCode != 200 {
+		return errInvalidToken
+	}
+	buf := make([]byte, rsp.ContentLength)
+	n, _ := rsp.Body.Read(buf)
```

Comment:
Why not just returning the error from rsp.Body.Read(buf) if there is any, instead of the content-length checking?